dotnet/src/SemanticKernel.Skills/Skills.Document/FileSystem/LocalFileSystemConnector.cs

```diff
@@ -72,4 +72,10 @@ public Task<bool> FileExistsAsync(string filePath, CancellationToken cancellatio
     {
         return Task.FromResult(File.Exists(Environment.ExpandEnvironmentVariables(filePath)));
     }
+
+    /// <inheritdoc/>
+    public Task<string[]> RecursivelyListFilesUnderDirectoryAsync(string directoryPath, CancellationToken cancellationToken = default)
+    {
+        return Task.FromResult(Directory.GetFiles(directoryPath, "*.*", SearchOption.AllDirectories));
```

Comment:
is there any risk of infinite loops with soft and hard links? hard links in particular are hard to detect, maybe we should define a max depth

Reply:
I'll test this out.